homeassistant/components/mychevy/sensor.py

```diff
@@ -160,6 +160,8 @@ def async_update_callback(self):
         """Update state."""
         if self._car is not None:
             self._state = getattr(self._car, self._attr, None)
+            if self._unit_of_measurement == "miles":
+                self._state = round(self._state)
```

Comment:
Shouldn't this be handled and fixed upstream? I highly doubt it would return 9 decimals from the actual car, so this is a bug in the upstream Python package, right?

Reply:
The upstream REST API from GM is returning a lot of decimals (which is bizarre, but :shrug: . Here is example data. 

```json
{"messages":[],"serverErrorMsgs":[],"data":{"dataAsOfDate":1615076067455,"batteryLevel":57.6,"chargeState":"not_charging","plugState":"unplugged","rateType":"OFFPEAK","voltage":0.0,"electricRange":92.58430731,"totalRange":92.58430731,"chargeMode":"IMMEDIATE","electricMiles":40313.2455002772,"gasMiles":0.0,"totalMiles":40313.2455002772,"percentageOnElectric":0.0,"fuelEconomy":0.0,"electricEconomy":18.4,"combinedEconomy":113.7732937971,"fuelUsed":0.0,"electricityUsed":11937.6,"estimatedGallonsFuelSaved":0.0,"estimatedCO2Avoided":0.0,"estimatedFullChargeBy":"11:00 PM ","gasFuelLevelPercentage":null,"electricRangeUOM":"mi","totalRangeUOM":null,"gasMilesUOM":null,"electricMilesUOM":null,"totalMilesUOM":null,"fuelEconomyUOM":null,"electricEconomyUOM":"kwh","combinedEconomyUOM":"mpge","fuelUsedUOM":null,"electricityUsedUOM":"kwh","estimatedGallonsFuelSavedUOM":null,"estimatedCO2AvoidedUOM":null,"gasRangeUOM":null,"gasRange":0.0}}
```

I think the change to way too many decimals coming from the API was recent. We can shield it in the upstream library, but realize this is what's coming straight from the car.

Reply:
In general we should round it as close to the display layer as possible so I think this is fine to do here since the data is actually coming from the api.